experimental/ActionPolyRing/src/Content.jl

```diff
@@ -0,0 +1,1162 @@
+end
+
+###############################################################################
+#
+#  Random generation
+#
+###############################################################################
+
+rand(rng::AbstractRNG, apr::ActionPolyRing, term_range::AbstractUnitRange{Int},
+     exp_bound::AbstractUnitRange{Int}, v...) = apr(rand(rng, __upr(apr), term_range, exp_bound, v...))
+
+rand(apr::ActionPolyRing, term_range, exp_bound, v...) = rand(Random.default_rng(), apr, term_range, exp_bound, v...)
+
+###############################################################################
+#
+#  Conformance test element generation
+#
+###############################################################################
+
+ConformanceTests.generate_element(R::ActionPolyRing{ZZRingElem}) = rand(R, 0:4, 0:10, -10:10)
+ConformanceTests.generate_element(R::ActionPolyRing{ZZModRingElem}) = rand(R, 0:4, 0:10, -10:10)
+
+###############################################################################
+#
+#  Misc 
+#
+###############################################################################
+
+symbols(apr::ActionPolyRing) = symbols(__upr(apr))[__perm_for_sort(apr)]
+
+canonical_unit(apre::ActionPolyRingElem) = canonical_unit(data(apre))
+
+###############################################################################
+#
+#  Aux action polynomial rings 
+#
+###############################################################################
+
+# =========================================
+# Getters for ring elements and rings
+# =========================================
+
+# Element getters
+data(dpre::Union{DifferencePolyRingElem, DifferentialPolyRingElem}) = dpre.p
+
+__is_perm_up_to_date(dpre::Union{DifferencePolyRingElem, DifferentialPolyRingElem}) = dpre.is_perm_up_to_date
+
+# Ring getters
+__upr(dpr::Union{DifferencePolyRing, DifferentialPolyRing}) = dpr.upoly_ring
+__jtv(dpr::Union{DifferencePolyRing, DifferentialPolyRing}) = dpr.jet_to_var::Dict{Tuple{Int, Vector{Int}}, elem_type(dpr)}
+__vtj(dpr::Union{DifferencePolyRing, DifferentialPolyRing}) = dpr.var_to_jet::Dict{elem_type(dpr), Tuple{Int, Vector{Int}}}
+__jtu_idx(dpr::Union{DifferencePolyRing, DifferentialPolyRing}) = dpr.jet_to_upoly_idx
+__are_perms_up_to_date(dpr::Union{DifferencePolyRing, DifferentialPolyRing}) = dpr.are_perms_up_to_date
+
+# =========================================
+# Permutation getters
+# =========================================
+
+function __perm_for_sort(dpr::Union{DifferencePolyRing, DifferentialPolyRing})
+  if __are_perms_up_to_date(dpr) && isdefined(dpr, :permutation)
+    return dpr.permutation
+  end
+  __set_perm_for_sort!(dpr)
+  return dpr.permutation
+end
+
+function __perm_for_sort_poly(dpre::Union{DifferencePolyRingElem, DifferentialPolyRingElem})
+  dpr = parent(dpre)
+  if __are_perms_up_to_date(dpr) && __is_perm_up_to_date(dpre)
+    return dpre.permutation
+  end
+  __set_perm_for_sort!(dpr)
+  __set_perm_for_sort_poly!(dpre)
```

Comment:
```suggestion
  if !__are_perms_up_to_date(dpr) || !__is_perm_up_to_date(dpre)
    __set_perm_for_sort!(dpr)
    __set_perm_for_sort_poly!(dpre)
  end
```
more readable IMO, but just my personal opinion. feel free to ignore